File: BringMeHome/BringMeHome.Services/Database/Shelter.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BringMeHome.Services.Database
{
        public class Shelter
        {
            [Key]
            public int ShelterID { get; set; }

            [Required, StringLength(100)]
            public string Name { get; set; }

            [StringLength(200)]
            public string Address { get; set; }

            public int CityID { get; set; }

            public int CantonID { get; set; }

            [StringLength(20)]
            public string ZipCode { get; set; }

            [StringLength(20)]
            public string Phone { get; set; }

            [EmailAddress, StringLength(100)]
            public string Email { get; set; }

            public int Capacity { get; set; }

            public int CurrentOccupancy { get; set; }

            [StringLength(200)]
            public string OperatingHours { get; set; }

            // Navigation properties
            [ForeignKey("CityID")]
            public virtual City City { get; set; }

            [ForeignKey("CantonID")]
            public virtual Canton Canton { get; set; }

            public virtual ICollection<Animal> Animals { get; set; }
            public virtual ICollection<Staff> Staff { get; set; }
        }
    }

File: bringmehomeUI/bringmehome_admin/lib/models/shelter.dart
class Shelter {
  final int shelterID;
  final String name;
  final String? address;
  final int cityID;
  final String? zipCode;
  final String? phone;
  final String? email;
  final int capacity;
  final int currentOccupancy;
  final String? operatingHours;

  Shelter({
    required this.shelterID,
    required this.name,
    this.address,
    required this.cityID,
    this.zipCode,
    this.phone,
    this.email,
    required this.capacity,
    required this.currentOccupancy,
    this.operatingHours,
  });

  factory Shelter.fromJson(Map<String, dynamic> json) {
    return Shelter(
      shelterID: json['shelterID'],
      name: json['name'],
      address: json['address'],
      cityID: json['cityID'],
      zipCode: json['zipCode'],
      phone: json['phone'],
      email: json['email'],
      capacity: json['capacity'],
      currentOccupancy: json['currentOccupancy'],
      operatingHours: json['operatingHours'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'shelterID': shelterID,
      'name': name,
      'address': address,
      'cityID': cityID,
      'zipCode': zipCode,
      'phone': phone,
      'email': email,
      'capacity': capacity,
      'currentOccupancy': currentOccupancy,
      'operatingHours': operatingHours,
    };
  }
}